### PR TITLE
Convert RulesLog to a logging service, first version. Issue #2404089

### PR DESCRIPTION
**Pull request by lussoluca**
Here there is a first attempt to convert the RulesLog class to a service. One thing I'm not sure is the use of Psr\Log\LogLevel levels because they're string constants and not integers, so I can't compare for level threshold.

### CODE REVIEW

File: tests/modules/rules_test/src/Plugin/Action/TestAction.php [hunk 1]
```diff
@@ -24,7 +24,8 @@ class TestAction extends RulesActionBase {
    * {@inheritdoc}
    */
   public function execute() {
-    RulesLog::logger()->log('action called');
+    $logger = \Drupal::service('rules.log');
```
Comment (fubhy) [inline tests/modules/rules_test/src/Plugin/Action/TestAction.php:27]: This should be injected with static ::create().

File: templates/rules_debug_element.html.twig [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="rules-debug-block">
```
Comment (fubhy) [inline templates/rules_debug_element.html.twig:1]: If I am reading this right this is a web profiler template. This should not be in rules but rather in the webprofiler module.

File: src/Engine/RulesLogInterface.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  /**
+   * Gets an array of logged messages.
+   */
+  public function get();
+
+  /**
+   * Renders the whole log.
+   */
+  public function render();
```
Comment (fubhy) [inline src/Engine/RulesLogInterface.php:53]: I think we should remove the render() part. The rules logger should simply extend AbstractLogger. Rendering should not be part of it but rather be outsourced into a separate service which can then be invoked from a specific controller which shows all log entries on a distinct page and can also be used to render the current rule log for the active request somewhere on the page.

File: src/Engine/RulesLog.php [hunk 4]
```diff
@@ -6,75 +6,89 @@
...
 
 /**
  * The rules default logging class.
  */
-class RulesLog {
-
-  const INFO  = 1;
-  const WARN  = 2;
-  const ERROR = 3;
+class RulesLog implements RulesLogInterface {
```
Comment (fubhy) [inline src/Engine/RulesLog.php:20]: RulesLog should extend AbstractLogger, no?

**Comment by fubhy** [comment]
I think this issue needs some fundamental conceptual planning and discussion for how we want RulesLogger to work. I think the PSR Logger (AbstractLogger) can be used directly (we need to consider a few things though). Most importantly, for allowing the same functionality we had in D7 (showing the debugger on the current page for the active request) we need to always pass a value to $context to identify the request so we can later retrieve the log entries for the current request easily. There is more to be considered though of course. I am up for discusson. You can find me on IRC or during the next Rules Hangout. Thanks for your work so far!

**Comment by M1r1k** [comment]
Futher development is here #170

**Comment by fubhy** [comment]
https://github.com/fago/rules/pull/170 got merged. Closing this one. Thanks for the effort (mentioned both of you in the merge commit message).

**Pull request closed without merging**